fvm/transactionInvoker.go

```diff
@@ -282,25 +218,92 @@ func (i TransactionInvoker) deductTransactionFees(
 
 // logExecutionIntensities logs execution intensities of the transaction
 func (i TransactionInvoker) logExecutionIntensities(
-	ctx Context,
 	txnState *state.TransactionState,
-	txHash string,
+	env environment.Environment,
 ) {
-	if ctx.Logger.Debug().Enabled() {
-		computation := zerolog.Dict()
-		for s, u := range txnState.ComputationIntensities() {
-			computation.Uint(strconv.FormatUint(uint64(s), 10), u)
-		}
-		memory := zerolog.Dict()
-		for s, u := range txnState.MemoryIntensities() {
-			memory.Uint(strconv.FormatUint(uint64(s), 10), u)
-		}
-		ctx.Logger.Info().
-			Uint64("ledgerInteractionUsed", txnState.InteractionUsed()).
-			Uint("computationUsed", txnState.TotalComputationUsed()).
-			Uint64("memoryEstimate", txnState.TotalMemoryEstimate()).
-			Dict("computationIntensities", computation).
-			Dict("memoryIntensities", memory).
-			Msg("transaction execution data")
+	if !env.Logger().Debug().Enabled() {
+		return
+	}
+
+	computation := zerolog.Dict()
+	for s, u := range txnState.ComputationIntensities() {
+		computation.Uint(strconv.FormatUint(uint64(s), 10), u)
+	}
+	memory := zerolog.Dict()
+	for s, u := range txnState.MemoryIntensities() {
+		memory.Uint(strconv.FormatUint(uint64(s), 10), u)
+	}
+	env.Logger().Info().
+		Uint64("ledgerInteractionUsed", txnState.InteractionUsed()).
+		Uint("computationUsed", txnState.TotalComputationUsed()).
+		Uint64("memoryEstimate", txnState.TotalMemoryEstimate()).
+		Dict("computationIntensities", computation).
+		Dict("memoryIntensities", memory).
+		Msg("transaction execution data")
+}
+
+func (i TransactionInvoker) normalExecution(
+	proc *TransactionProcedure,
+	txnState *state.TransactionState,
+	env environment.Environment,
+) (
+	computationUsed uint64,
+	memoryEstimate uint64,
+	modifiedSets programsCache.ModifiedSetsInvalidator,
+	err error,
+) {
+	rt := env.BorrowCadenceRuntime()
+	defer env.ReturnCadenceRuntime(rt)
+
+	err = rt.ExecuteTransaction(
+		runtime.Script{
+			Source:    proc.Transaction.Script,
+			Arguments: proc.Transaction.Arguments,
+		},
+		common.TransactionLocation(proc.ID))
+
+	// This will be used to charge fees.
+	computationUsed = env.ComputationUsed()
```

Comment:
note that i've also move these two line to up, before checking for execution error, since we need the computationUsed for fees deduction on the error path.